internal/store.go

```diff
@@ -738,18 +753,26 @@ func (s *Store[K, V]) Stats() Stats {
 	return newStats(s.policy.hits.Value(), s.policy.misses.Value())
 }
 
+// Close waits for all current read and write operations to complete,
+// then clears the hashmap and shuts down the maintenance goroutine.
+// After the cache is closed, Get will always return (nil, false),
+// and Set will have no effect.
+// For loading cache, Get will return ErrCacheClosed after closing.
 func (s *Store[K, V]) Close() {
-
 	for _, s := range s.shards {
```

Comment:
```suggestion
	for _, shard := range s.shards {
```

otherwise it gets confusing reading this, as `s` is already the Store itself.